lib/fiddle/struct.rb

```diff
@@ -94,33 +109,53 @@ def create(klass, types, members)
           define_method(name){ @entity[name] }
           define_method(name + "="){|val| @entity[name] = val }
         }
+        define_singleton_method(:malloc) do |func=nil|
+          struct = new(entity_class.malloc(types, func, size))
+          if block_given?
+            raise ArgumentError, 'a free function must be supplied to malloc when it is called with a block' unless func
+            begin
+              yield struct
+            ensure
+              struct.to_ptr.call_free
+            end
+          else
+            struct
+          end
+        end
       }
       size = klass.entity_class.size(types)
+      # we use eval here make size into a literal, for performance
```

Comment:
We can use `define_singleton_method` for `size` here because this method will not be called too much.

Reply:
Done.